apps/base-docs/docs/building-with-base/guides/cross-chain-with-ccip.md

```diff
@@ -0,0 +1,615 @@
+
+:::caution
+
+For instructions on how to get your private key from Coinbase Wallet, visit the [Coinbase Wallet documentation](https://docs.cloud.coinbase.com/wallet-sdk/docs/developer-settings#show-private-key). **It is critical that you do NOT commit this to a public repo**.
+
+:::
+
+To confirm that the wallet was imported as the `deployer` account in your Foundry project, run:
+
+```bash
+cast wallet list
+```
+
+### Setting up environment variables
+
+To setup your environment, create an `.env` file in the home directory of your project, and add the RPC URLs, [CCIP chain selectors](https://docs.chain.link/ccip/supported-networks/v1_2_0/testnet), [CCIP router addresses](https://docs.chain.link/ccip/supported-networks/v1_2_0/testnet), and [LINK token addresses](https://docs.chain.link/resources/link-token-contracts) for both Base Goerli and Optimism Goerli testnets:
+
+```bash
+BASE_GOERLI_RPC="https://goerli.base.org"
+OPTIMISM_GOERLI_RPC="https://goerli.optimism.io"
+
+BASE_GOERLI_CHAIN_SELECTOR=5790810961207155433
+OPTIMISM_GOERLI_CHAIN_SELECTOR=2664363617261496610
+
+BASE_GOERLI_ROUTER_ADDRESS="0x80AF2F44ed0469018922c9F483dc5A909862fdc2"
+OPTIMISM_GOERLI_ROUTER_ADDRESS="0xcc5a0B910D9E9504A7561934bed294c51285a78D"
+
+BASE_GOERLI_LINK_ADDRESS="0x6D0F8D488B669aa9BA2D0f0b7B75a88bf5051CD3"
+OPTIMISM_GOERLI_LINK_ADDRESS="0xdc2CC710e42857672E7907CF474a69B63B93089f"
+```
+
+Once the `.env` file has been created, run the following command to load the environment variables in the current command line session:
+
+```bash
+source .env
+```
+
+### Deploying the smart contracts
+
+With your contracts compiled and environment setup, you are ready to deploy the smart contracts.
+
+To deploy a smart contract using Foundry, you can use the `forge create` command. The command requires you to specify the smart contract you want to deploy, an RPC URL of the network you want to deploy to, and the account you want to deploy with.
+
+:::info
+
+Your wallet must be funded with ETH on the Base Goerli and Optimism Goerli to cover the gas fees associated with the smart contract deployment. Otherwise, the deployment will fail.
+
+To get testnet ETH for Base Goerli and Optimism Goerli, see the [prerequisites](#prerequisites).
+
+:::
+
+#### Deploying the Sender contract to Base Goerli
+
+To deploy the `Sender` smart contract to the Base Goerli testnet, run the following command:
+
+```bash
+forge create ./src/Sender.sol:Sender --rpc-url $BASE_GOERLI_RPC --constructor-args $BASE_GOERLI_ROUTER_ADDRESS $BASE_GOERLI_LINK_ADDRESS --account deployer
+```
+
+When prompted, enter the password that you set earlier, when you imported your wallet's private key.
+
+After running the command above, the contract will be deployed on the Base Goerli test network. You can view the deployment status and contract by using a [block explorer](/tools/block-explorers).
+
+#### Deploying the Receiver contract to Optimism Goerli
```

Comment:
These parts do make sense to me being here since they are unique to this article, having to deploy on multiple networks